native/data_dypes.md

```diff
@@ -0,0 +1,102 @@
+ - V8 : A Javascript engine developed by Google.
+
+
+#### JS Types
+
+The latest ECMAScript standard defines seven data types: (ref. MDN and contributors [2015] [Online] Available from [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Data_structures))
+
+    Six data types that are primitives:
+        Boolean
+        Null
+        Undefined
+        Number
+        String
+        Symbol (new in ECMAScript 6)
+    and 
+        Object
+
+Initial list of the corresponding native types are given below;
+
+```
+enum JSTypes {
+  Int32, // Number
+  Double, // Number
+  Boolean, // Boolean
+  String, // String
+  Object, // Object
+  Buffer, // Specialized Object
+  Undefined, // Undefined
+  Null, // Null
+  Error, // Specialized Object
+  Function // Callable Object
+};
+```
+
+`JSTypes type_;`
+
+Given list does not provide a definition for ObjectTemplate or FunctionTemplate V8 types. Assumption is that, these sub specializations can be provided via API methods. 
+
+`JS Engine` native type definition and / or behavior is engine and version specific. However, regardless from the implementation details, previously mentioned data types are available for all the mature JS engines.
+
+Following the **common** `JS engine` behavior, there are two important memory state a native type representation can be exist;
+
+ - Persistent / Rooted Variable
+ - Scoped / Temp Variable
+
+`Node` is not expected to manage the memory for JS Engine variable for any of the given options above. One of the [verified](https://github.com/jxcore/jxcore/blob/master/doc/native/Embedding_Basics.md) assumptions show that; a void ptr can point to a native JS Engine variable without dealing with its memory management.
+
+`void *data_;` -> JS Engine Variable
+
+Similar to local C,C++ variables, if `data_` holds a scope variable, it's value is not expected to be available after function's scope. Persistent or rooted variables are not restricted with function scope. However, they need to be freed manually.
+
+V8 API mentions a `weak-able` variable. This type of variable is not expected to be GC'ed by `JS engine` until there is no Javascript side variable referencing to it. 
+
+`JS engines` by default do not GC any active variable. As long as there is one another (non GC-able) variable holding a reference to the subject variable, it should be reachable. However, V8 also doesn't guarantee that the `native representation` of non-persistent but weakened variable will persist. As a result, from the native side scope/data point of view, `weak-able` variables do not bring a new condition or challenge. All the mature `JS Engines` provide somehow a similar functionality.
+
+V8 engine (similar to other mature `JS engines`) has a GC event mechanism on C,C++ side to tell a weak variable has finally / or about to be garbage collected. Although the timing of the GC needs a separate discussion, it has not much to do with data type abstraction subject. 
```

Comment:
node, nor any modules I'm aware of, attempt to resurrect and object just before GC. Though _when_ GC runs has one caveat. If the GC doesn't run weak callbacks until after the stack has unwound then something like the following would cause the system to explode:
```js
for (var i = 0; i < 1e7; i++)
  new Buffer(0xfffff);
```